### PR TITLE
Fix ui-select options placeholders for url-filter-applier

### LINKED ISSUE #1694: Used In Link: Asset title is not displayed in the applied filters section. Asset filter does not have the selected value
### Description (*)

See magento/adobe-stock-integration#1503: Add links to grids to the image details panel

### Preconditions
Have an image in media gallery that was inserted to page, block, category and product content

### Manual testing scenarios (*)
1. Open the Media Gallery
2. View details of the image
3. Click links in Used In section

### Expected Result
Corresponding grid is opened with the entities filtered by the image (asset), Asset filter is set and displayed correctly

### Actual Result
Asset title is not displayed in the applied filters section. Asset filter does not have the selected value
image

### LINKED ISSUE #1755: Tag isn`t added to second image in Edit mode
### Preconditions (*)

1. Install Magento 2.4-develop with Adobe Stock Integration and IMS
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset
3. Enhanced Media Gallery Enabled
4. Upload 2 images

### Steps to reproduce (*)

1. Go to Content - Media Gallery
2. Select first image
3. Click on three dots -> Edit
4. Add description
5. Add tag
6. Click on the "Cancel" button 
7. Select second image
8. Click on three dots -> Edit
9. Add description
10. Try to add tag

### Expected result (*)

Tag should be added successfully

### Actual result (*)

Tag isn`t added
Peek 2020-08-13 20-29

### PR DESCRIPTION
**Pull request by Nazar65**
### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration#1694: Used In Link: Asset title is not displayed in the applied filters section. Asset filter does not have the selected value
2. Fixes magento/adobe-stock-integration#1755: Tag isn`t added to second image in Edit mode

### Manual testing scenarios (*)
1. Open the Media Gallery
2. View details of the image
3. Click links in Used In section

### Expected Result
Corresponding grid is opened with the entities filtered by the image (asset), Asset filter is set and displayed correctly

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Nazar65** [comment]
@magenot run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

File: app/code/Magento/MediaGalleryUi/Ui/Component/Listing/Filters/Asset.php [hunk 1]
```diff
@@ -58,6 +93,61 @@ public function __construct(
...
+     * Prepare component configuration
+     *
+     * @return void
+     */
+    public function prepare()
+    {
+        $options = [];
+        $assetIds = [];
+
+        if (!isset($this->filterData[$this->getName()])) {
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryUi/Ui/Component/Listing/Filters/Asset.php]: please extract `getAssetIds` method to reduce the complexity

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @Nazar65 ! Please see my comment, the Asset filter readability can be improved with some refactoring.

Also please cover the View details links with MFTF test verifying the filter labels

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8019](https://jira.corp.magento.com/browse/ENGCOM-8019) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by Nazar65** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8019](https://jira.corp.magento.com/browse/ENGCOM-8019) has been created to process this Pull Request

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed
Corresponding grid is opened with the entities filtered by the image (asset), Asset filter is set and displayed correctly
Screenshot_107

Screenshot_108

Screenshot_109

Screenshot_110

Tag added to second image in Edit mode
Peek 2020-08-17 19-55

**Pull request merged**